### ISSUE TITLE
AWS Network ACL resource doesn't support * for rule_no

### ISSUE THREAD

**Issue by mcspadden** [hashicorp/terraform-provider-aws#21781]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version
Terraform CLI: `1.0.10`
Terraform AWS Provider: `~> 3.0`

### Affected Resource(s)

* [aws_network_acl](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/network_acl)

### Terraform Configuration Files
```
resource "aws_network_acl" "network-acl" {
  vpc_id = aws_vpc.vpc.id
  subnet_ids = [
    aws_subnet.subnet_public_01.id,
    aws_subnet.subnet_public_02.id
  ]

  egress {
    protocol   = "-1"
    rule_no    = 100
    action     = "allow"
    cidr_block = "0.0.0.0/0"
    from_port  = 0
    to_port    = 0
  }
  egress {
    protocol   = "-1"
    rule_no    = *            # "*"/-1 doesn't work either
    action     = "deny"
    cidr_block = "0.0.0.0/0"
    from_port  = 0
    to_port    = 0
  }
  ingress {
    protocol   = "-1"
    rule_no    = 100
    action     = "allow"
    cidr_block = "0.0.0.0/0"
    from_port  = 0
    to_port    = 0
  }
  ingress {
    protocol   = "-1"
    rule_no    = *            # "*"/-1 doesn't work either
    action     = "deny"
    cidr_block = "0.0.0.0/0"
    from_port  = 0
    to_port    = 0
  }
}
```

### Debug Output

[Debug Output](https://gist.github.com/mcspadden/3f04637de900a71b4e8382344b185029)

### Expected Behavior

Expected to place a `*` as the `rule_no` as shown here: https://docs.aws.amazon.com/vpc/latest/userguide/vpc-network-acls.html#custom-network-acl

### Actual Behavior

The following error:
`
Expected the start of an expression, but found an invalid expression token.
`

### Steps to Reproduce

1. `terraform init` containing the above resource with rule_no defined as `*`, `"*"`, or `-1`.

### Important Factoids

Basically just using the network ACL as a passthrough at the VPC level for the time being.

### References
Couldn't find another similar issue here

**Comment by mcspadden** [comment]
This is actually created by default for all `aws_network_acl`'s on AWS side so closing this issue as there is not a fix needed.
